libs/methodologies/bold/rule-processors/mass-id/src/rewards-distribution/rewards-distribution.processor.ts

```diff
@@ -0,0 +1,413 @@
+import {
+  getOrDefault,
+  isNil,
+  isNonEmptyArray,
+} from '@carrot-fndn/shared/helpers';
+import { getEventAttributeValue } from '@carrot-fndn/shared/methodologies/bold/getters';
+import {
+  type DocumentQuery,
+  DocumentQueryService,
+} from '@carrot-fndn/shared/methodologies/bold/io-helpers';
+import {
+  MASS_ID,
+  METHODOLOGY_DEFINITION,
+} from '@carrot-fndn/shared/methodologies/bold/matchers';
+import {
+  eventLabelIsAnyOf,
+  isActorEvent,
+} from '@carrot-fndn/shared/methodologies/bold/predicates';
+import {
+  type Document,
+  DocumentEventAttributeName,
+  DocumentEventAttributeValue,
+  MassIdOrganicSubtype,
+  RewardsDistributionActorType,
+} from '@carrot-fndn/shared/methodologies/bold/types';
+import { mapDocumentReference } from '@carrot-fndn/shared/methodologies/bold/utils';
+import { mapToRuleOutput } from '@carrot-fndn/shared/rule/result';
+import {
+  type RuleInput,
+  type RuleOutput,
+  RuleOutputStatus,
+} from '@carrot-fndn/shared/rule/types';
+import { MethodologyDocumentEventLabel } from '@carrot-fndn/shared/types';
+import { BigNumber } from 'bignumber.js';
+import { is } from 'typia';
+
+import {
+  REQUIRED_ACTOR_TYPES,
+  REWARDS_DISTRIBUTION,
+  REWARDS_DISTRIBUTION_BY_WASTE_TYPE,
+  REWARDS_DISTRIBUTION_CRITERIA,
+} from './rewards-distribution.constants';
+import { RewardsDistributionProcessorErrors } from './rewards-distribution.errors';
+import {
+  checkIfHaulerActorExists,
+  getActorsByType,
+  mapActorReward,
+  mapMassIdRewards,
+} from './rewards-distribution.helpers';
+import {
+  type ActorMassIdPercentageInputDto,
+  type ActorReward,
+  type RewardsDistributionActor,
+  type RewardsDistributionActorTypePercentage,
+} from './rewards-distribution.types';
+
+const { UNIDENTIFIED } = DocumentEventAttributeValue;
+const { WASTE_ORIGIN } = DocumentEventAttributeName;
+const { WASTE_GENERATOR } = MethodologyDocumentEventLabel;
+
+BigNumber.config({ DECIMAL_PLACES: 10, ROUNDING_MODE: BigNumber.ROUND_DOWN });
+
+const LARGE_SOURCE_COMPANY_DISCOUNT = 0.5;
+
+export class RewardsDistributionProcessor extends RuleDataProcessor {
+  private checkIfHasRequiredActorTypes = ({
+    actors,
+    documentId,
+    requiredActorTypes,
+  }: {
+    actors: RewardsDistributionActor[];
+    documentId: string;
+    requiredActorTypes: RewardsDistributionActorType[];
+  }) => {
+    const missingRequiredActors = requiredActorTypes.filter(
+      (requiredActorType) =>
+        !actors.some((actor) => actor.type === requiredActorType),
+    );
+
+    if (isNonEmptyArray(missingRequiredActors)) {
+      throw this.errorProcessor.getKnownError(
+        this.errorProcessor.ERROR_MESSAGE.MISSING_REQUIRED_ACTORS(
+          documentId,
+          missingRequiredActors,
+        ),
+      );
+    }
+  };
+
+  readonly errorProcessor = new RewardsDistributionProcessorErrors();
+
+  private checkIfWasteOriginIsNotIdentified(document: Document): boolean {
```

Comment:
For better reading, I think it is better to avoid using `not` is the function name.
```suggestion
  private isWasteOriginIdentified(document: Document): boolean {
```

Reply:
https://github.com/carrot-foundation/methodology-rules/pull/216/commits/33b81170092a1f07177ba68402d9a0f35cd2eefe